packages/ui/src/components/EasySetup/IdentitySetIdentity.tsx

```diff
@@ -0,0 +1,127 @@
+import { Grid } from '@mui/material'
+import { styled } from '@mui/material/styles'
+import { SubmittableExtrinsic } from '@polkadot/api/types'
+import { ISubmittableResult } from '@polkadot/types/types'
+import { useCallback, useEffect, useState } from 'react'
+import { useApi } from '../../contexts/ApiContext'
+import { TextFieldStyled } from '../library'
+import { useIdentity } from '../../hooks/useIdentity'
+
+interface Props {
+  className?: string
+  from: string
+  onSetExtrinsic: (ext?: SubmittableExtrinsic<'promise', ISubmittableResult>) => void
+}
+
+interface IdentityFields {
+  display: string | undefined
+  legal: string | undefined
+  web: string | undefined
+  riot: string | undefined
+  email: string | undefined
+  image: string | undefined
+  twitter: string | undefined
+}
+
+const getRawOrNone = (val: string | undefined) => {
+  return val
+    ? {
+        Raw: val
+      }
+    : { none: null }
+}
+const getExtrinsicsArgs = (identity: IdentityFields) => {
+  const { legal, display, email, image, riot, twitter, web } = identity
+
+  return {
+    additional: [],
+    display: getRawOrNone(display),
+    legal: getRawOrNone(legal),
+    web: getRawOrNone(web),
+    riot: getRawOrNone(riot),
+    email: getRawOrNone(email),
+    pgpFingerprint: null,
+    image: getRawOrNone(image),
+    twitter: getRawOrNone(twitter)
+  }
+}
+
+const IdentitySetIdentity = ({ className, onSetExtrinsic, from }: Props) => {
+  const { api, chainInfo } = useApi()
+  const [identity, setIdentity] = useState<IdentityFields | undefined>()
```

Comment:
Can we manage this state inside useIdentity?

Reply:
I don't understand what you mean. This is the state of the specific fields. This isn't directly related to the identity from the chain. I can change the name to `identityFields` if that's more clear.

Reply:
To rename to identityFields is a good idea, as I was confused with the hook name which has the same name